### ISSUE TITLE
Request for add configurations on template and interfaces resources

### ISSUE THREAD

**Issue by gsantos1979** [CiscoDevNet/terraform-provider-iosxe#10]
Please is it possible to add the options
- "auto qos voip trust"
- "switchport trunk allowed vlan all | none"
- "spanning-tree guard root" 
on templates and interfaces resources, once is allowed at CLI:

```
sw1(config-if)#template UPLINK-SWITCHES
sw1(config-template)#switchport trunk allowed vlan ?
  WORD    VLAN IDs of the allowed VLANs when this port is in trunking mode
  add     add VLANs to the current list
  all     all VLANs
  except  all VLANs except the following
  none    no VLANs
  remove  remove VLANs from the current list

sw1(config-template)#trust ?
  cos      cos
  device   trusted device class
  dscp     dscp
  ip-prec  ip precedence
  up       user priority

sw1(config-template)#trust device ?
  cisco-phone   Cisco IP Phone
  cts           CTS
  ip-camera     IPVSC
  media-player  DMP

sw1(config-template)#trust cos ?
  <cr>  <cr>

sw1(config-template)#spanning-tree guard ?
  loop  Set guard mode to loop guard on interface
  none  Set guard mode to none
  root  Set guard mode to root guard on interface
```

The idea is configure generic templates for some types of interfaces (uplinks, access trunk, access edge, routers) and configure the allowed vlans directly on interfaces.

Thank you.

**Comment by danischm** [comment]
https://github.com/CiscoDevNet/terraform-provider-iosxe/commit/892c60179091aa2197dbf57e18670e3edadce19a

**Comment by danischm** [comment]
I have added what is available in the XE YANG models. Some of the template configurations are not modelled and can therefore not be added.

**Comment by danischm** [comment]
@sdeweese @robertcsapo @jeremycohoe

**Comment by danischm** [comment]
available in v0.3.1

**Comment by gsantos1979** [comment]
Thank you @danischm
